Convert __constants__ attribute in model to a set to be consistent

SplitInfinity opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#60003 [JIT] Handle modules that already have __constants__**

**Summary**
`infer_concrete_type_builder` in `_recursive.py` assumes `__constants__`
is a `set` if it exists as an attribute on the module being scripted.
Instead, it should create a set out of whatever `__constants__` is.

**Test Plan**
Ran code from the issue.

**Fixes**
This commit fixes #59947.

Differential Revision: [D29174243](https://our.internmc.facebook.com/intern/diff/D29174243)

Commit:
[JIT] Handle modules that already have __constants__

**Summary**
`infer_concrete_type_builder` in `_recursive.py` assumes `__constants__`
is a `set` if it exists as an attribute on the module being scripted.
Instead, it should create a set out of whatever `__constants__` is.

**Test Plan**
Ran code from the issue.

**Fixes**
This commit fixes #59947.

[ghstack-poisoned]

SplitInfinity commented:
I would like to add the code from the issue as a unit test but I don't want to add to the mess that is `test_jit.py`. Any suggestions as to where this test should go?

gmagogsfm commented:
> I would like to add the code from the issue as a unit test but I don't want to add to the mess that is `test_jit.py`. Any suggestions as to where this test should go?

None of the existing tests look good, maybe start a new file?

SplitInfinity commented:
> > I would like to add the code from the issue as a unit test but I don't want to add to the mess that is `test_jit.py`. Any suggestions as to where this test should go?
> 
> None of the existing tests look good, maybe start a new file?

What should it be called?

Commit:
Update on "[JIT] Handle modules that already have __constants__"


**Summary**
`infer_concrete_type_builder` in `_recursive.py` assumes `__constants__`
is a `set` if it exists as an attribute on the module being scripted.
Instead, it should create a set out of whatever `__constants__` is.

**Test Plan**
Ran code from the issue.

**Fixes**
This commit fixes #59947.

[ghstack-poisoned]

Commit:
Update on "[JIT] Handle modules that already have __constants__"


**Summary**
`infer_concrete_type_builder` in `_recursive.py` assumes `__constants__`
is a `set` if it exists as an attribute on the module being scripted.
Instead, it should create a set out of whatever `__constants__` is.

**Test Plan**
Ran code from the issue.

**Fixes**
This commit fixes #59947.

[ghstack-poisoned]

SplitInfinity commented:
@splitinfinity has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29174243).
